docs/docs/adding-an-rss-feed.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Adding an RSS Feed
+---
+
+## What is an RSS feed?
+
+An [RSS Feed](https://en.wikipedia.org/wiki/RSS) is a standard XML file listing a website’s content in a subscribable format, allowing readers to consume your content in news aggregators, also called feed reader apps.
+
+Think of it as a syndicated distribution channel for your site's content.
+
+## Install
+
+To generate an RSS feed, you can use the [`gatsby-plugin-feed`](/packages/gatsby-plugin-feed/) package. To install this package, run the following command:
+
+```sh
+npm install --save gatsby-plugin-feed
+```
+
+## How to use [gatsby-plugin-feed](/packages/gatsby-plugin-feed/)
+
+Once installation is complete, you can now add this plugin to your site's config file, like so:
+
+```js:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    siteUrl: `https://www.example.com`,
+  },
+  plugins: [`gatsby-plugin-feed`],
+}
+```
+
+Here's an example of how you could implement this plugin with Markdown, but for other sources, you will need a way to uniquely identify content--typically a URL or slug.
+
+```js:title=gatsby-node.js
+const { createFilePath } = require(`gatsby-source-filesystem`)
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+  // highlight-next-line
+  if (node.internal.type === `MarkdownRemark`) {
+    const value = createFilePath({ node, getNode })
+    createNodeField({
+      name: `slug`,
+      node,
+      value,
+    })
+  }
+}
+```
+
+Next run a build (`npm run build`) since the RSS feed generation will only happen for production builds. By default, the generated RSS feed path is `/rss.xml`, but the plugin exposes options to configure this default functionality.
+
+For basic setups with Markdown content like the [gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog), that's all you need! However, you can craft a custom RSS feed schema using custom code in your `gatsby-node.js` and `gatsby-config.js` files.
+
+## Customizing the RSS feed plugin
+
+Your content might not fit neatly into the blog-starter scenario, for various reasons like:
+
+- Your content isn't in Markdown so the plugin doesn't know about it
+- Your Markdown files have dates in the filenames, for which the slug URLs cause 404s
+
+The good news is you can accommodate these scenarios and more in `gatsby-config.js` and `gatsby-node.js`.
+
+To customize the default feed schema (a.k.a. structure) output by the plugin to work with your website's content, you can start with the following code:
+
+```json:title=gatsby.config.js
```

Comment:
This is js!

Would generally recommend doing something like:

````md
```js:title=gatsby-config.js
module.exports = {
 plugins: [
   // your code block
 ]
}
```
````

just so we can show it as real-world as possible!

Reply:
Done!